wait/member.go

```diff
@@ -29,30 +29,56 @@ func NewMemberAwaitility(a *Awaitility) *MemberAwaitility {
 		}}
 }
 
+// UserAccountWaitCriterion a function to check that a user account has the expected condition
+type UserAccountWaitCriterion func(a *MemberAwaitility, ua *toolchainv1alpha1.UserAccount) bool
+
+// UntilUserAccountHasSpec returns a `UserAccountWaitCriterion` which checks that the given
+// USerAccount has the expected spec
+func UntilUserAccountHasSpec(expected toolchainv1alpha1.UserAccountSpec) UserAccountWaitCriterion {
+	return func(a *MemberAwaitility, ua *toolchainv1alpha1.UserAccount) bool {
+		return reflect.DeepEqual(ua.Spec, expected)
+	}
+}
+
+// UntilUserAccountHasConditions returns a `UserAccountWaitCriterion` which checks that the given
+// USerAccount has exactly all the given status conditions
+func UntilUserAccountHasConditions(conditions ...toolchainv1alpha1.Condition) UserAccountWaitCriterion {
+	return func(a *MemberAwaitility, ua *toolchainv1alpha1.UserAccount) bool {
+		if test.ConditionsMatch(ua.Status.Conditions, conditions...) {
+			a.T.Logf("status conditions match in UserAccount '%s`", ua.Name)
+			return true
+		}
+		a.T.Logf("waiting for correct status condition of UserAccount '%s`", ua.Name)
```

Comment:
I see that the tests are failing, but it's hard to get info what is wrong there. So I was thinking about adding an additional information to the log:
```suggestion
		a.T.Logf("waiting for correct status conditions [%+v] of UserAccount '%s`, the actual are: [%+v]", conditions, ua.Name, ua.Status.Conditions)
```